lib/puppet/pops/binder/producers.rb

```diff
@@ -456,7 +456,7 @@ def producer(scope)
     # @api private
     #
     def internal_produce(scope, *args)
-      producer() unless value_producer
```

Comment:
Still present in master: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/pops/binder/producers.rb#L459